File: src/main/java/com/barbaros/expense/controller/ExpenseController.java
package com.barbaros.expense.controller;

import com.barbaros.expense.dto.ExpenseDTO;
import com.barbaros.expense.service.ExpenseService;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/expenses")
public class ExpenseController {

    private final ExpenseService expenseService;

    public ExpenseController(ExpenseService expenseService) {
        this.expenseService = expenseService;
    }

    @PostMapping
    public ExpenseDTO createExpense(@RequestBody ExpenseDTO expenseDTO) {
        return expenseService.createExpense(expenseDTO);
    }

    @GetMapping
    public List<ExpenseDTO> getAllExpenses() {
        return expenseService.getAllExpenses();
    }

    @GetMapping("/{id}")
    public ExpenseDTO getExpenseById(@PathVariable Long id) {
        return expenseService.getExpenseById(id);
    }
}

File: src/main/kotlin/com/barbaros/expense/controller/ExpenseController.kt
package com.barbaros.expense.controller

import com.barbaros.expense.dto.ExpenseDTO
import com.barbaros.expense.exception.ResourceNotFoundException
import com.barbaros.expense.service.ExpenseService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/expenses")
class ExpenseController(private val expenseService: ExpenseService) {

    @PostMapping
    fun createExpense(@RequestBody expenseDTO: ExpenseDTO): ExpenseDTO {
        return expenseService.createExpense(expenseDTO)
    }

    @GetMapping
    fun getAllExpenses(): List<ExpenseDTO> {
        return expenseService.getAllExpenses()
    }

    @GetMapping("/{id}")
    fun getExpenseById(@PathVariable id: Long): ExpenseDTO {
        return expenseService.getExpenseById(id)
            ?: throw ResourceNotFoundException("Expense with ID $id not found")
    }
}
